arwen/contexts/output.go

```diff
@@ -387,6 +376,39 @@ func (context *outputContext) TransferESDT(
 	return gasRemaining, nil
 }
 
+func (context *outputContext) getOutputTransferDataFromESDTTransfer(
+	transfers []*vmcommon.ESDTTransfer,
+	vmOutput *vmcommon.VMOutput,
+	sameShard bool,
+	destination []byte,
+) []byte {
+
+	if len(transfers) == 1 && transfers[0].ESDTTokenNonce == 0 {
+		return []byte(vmcommon.BuiltInFunctionESDTTransfer + "@" + hex.EncodeToString(transfers[0].ESDTTokenName) + "@" + hex.EncodeToString(transfers[0].ESDTValue.Bytes()))
+	}
+
+	if !sameShard {
+		outTransfer, ok := vmOutput.OutputAccounts[string(destination)]
+		if ok && len(outTransfer.OutputTransfers) == 1 {
+			return outTransfer.OutputTransfers[0].Data
+		}
+	}
+
+	if len(transfers) == 1 {
+		data := []byte(vmcommon.BuiltInFunctionESDTNFTTransfer + "@" + hex.EncodeToString(transfers[0].ESDTTokenName) +
+			"@" + hex.EncodeToString(big.NewInt(0).SetUint64(transfers[0].ESDTTokenNonce).Bytes()) + "@" + hex.EncodeToString(transfers[0].ESDTValue.Bytes()))
+		data = append(data, []byte("@"+hex.EncodeToString(destination))...)
```

Comment:
Format as one `append` per line, for readability.

Reply:
reformatted - append was not needed